beacon-chain/rpc/beacon/attestations.go

```diff
@@ -113,12 +115,29 @@ func (bs *Server) ListAttestations(
 	}, nil
 }
 
-// StreamAttestations to clients every single time a new attestation is received.
-// TODO(#4184): Implement.
+// StreamAttestations to clients at the end of every slot. This method retrieves the
+// aggregated attestations currently in the pool at the start of a slot and sends
+// them over a gRPC stream.
 func (bs *Server) StreamAttestations(
-	_ *ptypes.Empty, _ ethpb.BeaconChain_StreamAttestationsServer,
+	_ *ptypes.Empty, stream ethpb.BeaconChain_StreamAttestationsServer,
 ) error {
-	return status.Error(codes.Unimplemented, "Not yet implemented")
+	genesisTime := bs.GenesisTimeFetcher.GenesisTime()
+	st := slotutil.GetSlotTicker(genesisTime, params.BeaconConfig().SecondsPerSlot)
```

Comment:
one concern for this, what if I call this method a large number of slots after genesis; ex: 100,000.
We need to be ignore all the ticks that were meant for the past.Otherwise the ticker will pretty much spam this loop.

Reply:
Confirmed with the ticker code it does not loop based on the time since genesis, it merely aligns the time with the current slot and ticks every time a new slot happens